src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -201,57 +201,52 @@ private LocalDefinition EmitSequenceAddress(BoundSequence sequence, AddressKind
 
             // when a sequence is happened to be a byref receiver
             // we may need to extend the life time of the target until we are done accessing it
-            // {.v ; v = Foo(); v}.Bar()     // v should be released after Bar() is over.
+            // {.v ; v = Foo(); v}.Bar()     // v should be released only after Bar() is done.
             LocalSymbol doNotRelease = null;
             if (tempOpt == null)
             {
-                BoundLocal referencedLocal = DigForLocal(sequence.Value);
-                if (referencedLocal != null)
+                doNotRelease = DigForValueLocal(sequence);
+                if (doNotRelease != null)
                 {
-                    doNotRelease = referencedLocal.LocalSymbol;
-                }
-            }
-
-            if (hasLocals)
-            {
-                _builder.CloseLocalScope();
-
-                foreach (var local in sequence.Locals)
-                {
-                    if (local != doNotRelease)
-                    {
-                        FreeLocal(local);
-                    }
-                    else
-                    {
-                        tempOpt = GetLocal(doNotRelease);
-                    }
+                    tempOpt = GetLocal(doNotRelease);
                 }
             }
 
+            FreeLocals(sequence, doNotRelease);
             return tempOpt;
         }
 
-        private BoundLocal DigForLocal(BoundExpression value)
+        // if sequence value is a local scoped to the sequence, return that local
+        private LocalSymbol DigForValueLocal(BoundSequence topSequence)
+        {
+            return DigForValueLocal(topSequence, topSequence.Value);
+        }
+
+        private LocalSymbol DigForValueLocal(BoundSequence topSequence, BoundExpression value)
         {
             switch (value.Kind)
             {
                 case BoundKind.Local:
                     var local = (BoundLocal)value;
-                    if (local.LocalSymbol.RefKind == RefKind.None)
+                    var symbol = local.LocalSymbol;
+                    if (topSequence.Locals.Contains(symbol))
                     {
-                        return local;
+                        return symbol;
                     }
                     break;
 
                 case BoundKind.Sequence:
-                    return DigForLocal(((BoundSequence)value).Value);
+                    return DigForValueLocal(topSequence, ((BoundSequence)value).Value);
 
                 case BoundKind.FieldAccess:
                     var fieldAccess = (BoundFieldAccess)value;
                     if (!fieldAccess.FieldSymbol.IsStatic)
                     {
-                        return DigForLocal(fieldAccess.ReceiverOpt);
+                        var receiver = fieldAccess.ReceiverOpt;
+                        if (!receiver.Type.IsReferenceType)
```

Comment:
no, since the field is not static. (and since we are emitting, the code has no semantical errors)